README.md

```diff
@@ -1,36 +1,70 @@
+[![Open in Visual Studio Code](https://classroom.github.com/assets/open-in-vscode-2e0aaae1b6195c2367325f4f02e2d04e9abb55f0b24a779b69b11b9e10269abc.svg)](https://classroom.github.com/online_ide?assignment_repo_id=15555388&assignment_repo_type=AssignmentRepo)
 # SE_Day1
 Software Engineering Day1 Assignment
 
 #Part 1: Introduction to Software Engineering
 
 Explain what software engineering is and discuss its importance in the technology industry.
 
+software engineering is the systematic application of engineering principles, methods and tools to the development and maintenance of high-quality software principles.
+software engineering helps in the creating of software applications and systems that power various aspects of modern life such as transport and logistics, healthcare, banking and finance and etc.
+
 
 Identify and describe at least three key milestones in the evolution of software engineering.
 
+three key milestones in the evolution of software engineering are: 
+(i) the development of programming languages eg Fortan,C
+(ii) the establishemetn of software engineering as a discpline in the 1960s
+(iii) the rise of agile methodologies in the 2000s
 
 List and briefly explain the phases of the Software Development Life Cycle.
 
+(1)Requirements - gathering user needs and documenting system requirements 
+(2)Design -creating high level and detailed designs of the software architecture and user interface
```

Comment:
- Integrated Development Environments (IDEs): Software suites that provide comprehensive tools for writing, debugging, and testing code (e.g., Visual Studio, Eclipse, IntelliJ IDEA).
  - Version Control Systems (VCS): Software tools for tracking changes to source code and coordinating work among team members (e.g., Git, Subversion). 
  
  Therefore, improves collaboration and productivity in software development

Reply:
Software Engineer - Design ,develop and maintain software applications.
Quality Assurance - idnetifying defects and ensuring the software runs as intended.
Project Manager -oversee planning ,execution , and completion of  a software.

Reply:
importances of an ide and vcs

They enhance productivity, streamline workflows, and improve code quality

examples of ides
vscode
intellij
eclipse

vcs
github
gitlab
aws commit

Reply:
IDEs are comprehensive software applications that provide developers with tools to write, edit, compile, and debug code efficiently. 
examples
visual studio
eclopse
intelliJ IDEA
VCS are tools that help manage changes to source code over time, providing a framework for collaboration and code management. 
examples
Git
SVN

Reply:
Integrated Development Environments (IDEs) streamline software development by combining tools like code editors, debuggers, and compilers into one platform. Examples include VS Code and IntelliJ IDEA.

Version Control Systems (VCS) track code changes, enable collaboration, and ensure safe code management. Tools like Git and SVN help developers work together, manage versions, and roll back changes if needed.

Both are key to improving efficiency and maintaining control in the development process.